docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,263 @@
+---
+layout: global
+title: TRANSFORM
+displayTitle: TRANSFORM
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specifies a Hive-style transform (SELECT TRANSFORM/MAP/REDUCE)
+query specification to transform the input by running a specified script. Users can
+plug in their own custom mappers and reducers in the data stream by using features natively supported
+in the Spark SQL. e.g. in order to run a custom mapper script `map_script` and a custom
+reducer script `reduce_script` the user can issue the following command which uses the TRANSFORM
+clause to embed the mapper and the reducer scripts.
+
+In default format, columns will be transformed to `STRING` and delimited by tabs before feeding
+to the user script, Similarly, all `NULL` values will be converted to the literal string `\N` in order to
+differentiate `NULL` values from empty strings. The standard output of the user script will be treated as
+TAB-separated STRING columns, any cell containing only `\N` will be re-interpreted as a `NULL`, and then the
+resulting STRING column will be cast to the data type specified in the table declaration in the usual way.
+User scripts can output debug information to standard error which will be shown on the task detail page on hadoop.
+These defaults can be overridden with `ROW FORMAT DELIMITED`.
+
+### Syntax
+
+```sql
+SELECT [ TRANSFORM ( namedExpressionSeq ) | MAP namedExpressionSeq | REDUCE namedExpressionSeq ]
+    [ inRowFormat ]
+    [ RECORDWRITER recordWriter_class ]
+    USING script [ AS ( [ col_name [ col_type ]] [ , ... ] ) ]
+    [ outRowFormat ]
+    [ RECORDREADER recordReader_class ]
+    FROM { from_item [ , ... ] }
```

Comment:
I think we don't need `FROM` here because this is described in the doc of `SELECT`: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select.html